### PR TITLE
Split keyword definitions from interpreter into their own module

### PR DESCRIPTION
**Pull request by dcbaker**
This also fixes the type checking of these keyword definitions.

@xclaesse, this should fix the issues you're seeing if you rebase on this

### CODE REVIEW

File: mesonbuild/interpreterbase/decorators.py [hunk 1]
```diff
@@ -323,7 +323,7 @@ def __init__(self, name: str, types: T.Union[T.Type[_T], T.Tuple[T.Type[_T], ...
                  deprecated: T.Optional[str] = None,
                  deprecated_values: T.Optional[T.Dict[str, str]] = None,
                  validator: T.Optional[T.Callable[[_T], T.Optional[str]]] = None,
-                 convertor: T.Optional[T.Callable[[_T], TYPE_var]] = None,
+                 convertor: T.Optional[T.Callable[[_T], object]] = None,
```
Comment (mensinda) [inline mesonbuild/interpreterbase/decorators.py:326]: Is there a specific reason for why this is required or is this is a debugging leftover?

**Comment by xclaesse** [comment]
LGTM when CI is done. Thanks!

File: mesonbuild/interpreterbase/decorators.py (hunk 1 above)
Comment (dcbaker) [reply mesonbuild/interpreterbase/decorators.py:326]: No, it's needed. The convertor function is in no way bound to return a TYPE_var, in fact of the three uses 2 don't return a TYPE_var. One of them converts the `bool` of the `native` keyword argument into a `MachineChoice`, the second converts the `install_mode` keyword argument into a `FileMode` object.

Comment (mensinda) [reply mesonbuild/interpreterbase/decorators.py:326]: Ok, fair enough.

**Review by mensinda — APPROVED** [review]
LGTM, when the CI is green.

**Pull request merged**